### PR TITLE
1545/lat-long draggable map

### PR DESCRIPTION
**Pull request by emilyjablonski**
# Pull Request Template

## Issue

Addresses #1545

- [x] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

Adds the map to listings management. When a user enters a full address, the map will render with automatic pin positioning. If the user wants to use a custom pin position they can select the custom radio button and the pin becomes draggable. When the listing is saved, the lat/long of the pin is saved as with the address.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Prototype/POC (not to merge)
- [ ] This change is a refactor/address technical debt
- [ ] This change requires a documentation update
- [ ] This change requires a SQL Script

## How Can This Be Tested/Reviewed?

Add a new listing in partners and play around with toggling automatic/custom positioning and previewing the listing.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [ ] I have updated the changelog to include a description of my changes

### CODE REVIEW

File: ui-components/src/page_components/listing/listing_sidebar/SidebarAddress.tsx [hunk 1]
```diff
@@ -1,13 +1,12 @@
 import * as React from "react"
 import ReactDOMServer from "react-dom/server"
 import { Icon, IconFillColors } from "../../../icons/Icon"
-import { Address } from "@bloom-housing/backend-core/types"
```
Comment (emilyjablonski) [inline ui-components/src/page_components/listing/listing_sidebar/SidebarAddress.tsx:4]: These changes were not necessary, just removing references to our backend from ui-components when I'm touching these components

File: sites/partners/src/listings/PaperListingForm/sections/BuildingDetails.tsx [hunk 2]
```diff
@@ -8,13 +8,116 @@ import {
...
+      buildingAddress.zipCode.length >= 5
+    )
+  }
+
+  let geocodingClient
+  if (process.env.mapBoxToken || process.env.MAPBOX_TOKEN) {
+    geocodingClient = GeocodeService({
+      accessToken: process.env.mapBoxToken || process.env.MAPBOX_TOKEN,
+    })
+  }
```
Comment (seanmalbert) [inline sites/partners/src/listings/PaperListingForm/sections/BuildingDetails.tsx:82]: Can you wrap this with a useEffect, so we only create the client once. I think you'll either have to move `let geocodingClient` above the component or use useState to set it.

File: sites/partners/src/listings/PaperListingForm/sections/BuildingDetails.tsx [hunk 3]
```diff
@@ -8,13 +8,116 @@ import {
...
+
+  const getNewLatLong = () => {
```
Comment (seanmalbert) [inline sites/partners/src/listings/PaperListingForm/sections/BuildingDetails.tsx:84]: I think we may want to [debounce](https://lodash.com/docs/#debounce) this function, so we're not calling it on every input change.

**Review by seanmalbert — CHANGES_REQUESTED** [review]
Hey @emilyjablonski ,
This looks great and thank you for the UI Component updates too! The only requests I have are around creating the map client and how often we call forwardGeocode. Please see my comments on the code.

File: sites/partners/src/listings/PaperListingForm/sections/BuildingDetails.tsx (hunk 3 above)
Comment (emilyjablonski) [reply sites/partners/src/listings/PaperListingForm/sections/BuildingDetails.tsx:91]: debouncing 👍  had lots of bugs w the lodash flavor so just went with a set/clear timeout

**Review by seanmalbert — APPROVED** [review]
Thanks for the updates, @emilyjablonski . I bumped the timeout up to 1 second, since I think we can get away with that.

**Pull request merged**